### DISCUSSION TITLE
Access to environment variables in Kong Serverless Functions plugins

### DISCUSSION THREAD

**Discussion by anup-krai** [Kong/kong#10194] [Help] [answered]
I am trying out Kong Serverless Functions plugins and in Kong docs it is mentioned that -
`You can't access or change global values # such as "kong.configuration.pg_password"`
[Reference here](https://github.com/Kong/kong/blob/master/kong.conf.default#L1752)

but when I try to access the environment variables its coming up without any issues. Below are the variables which I tried to print in post-function  -
```
kong.log.err("pg_password - ",kong.configuration.pg_password)
kong.log.err("untrusted_lua - ",kong.configuration.untrusted_lua)
kong.log.err("untrusted_lua_sandbox_environment - ",table.concat(kong.configuration.untrusted_lua_sandbox_environment))
```
Below is the output for this  -
```
[post-function] pg_password - ****. (this is removed form here but this is also getting printed)
[post-function] untrusted_lua - sandbox
[post-function] untrusted_lua_sandbox_environment - nil
```

Here I have not added any environment variable in `untrusted_lua_sandbox_environment` but still all are getting printed.

Is there any way where we can prevent the access to these environment variables in sandboxed environment?

**Comment by anup-krai** [comment]
Hi Team, 

Can you please let me know if this is working as expected?

**Comment by hanshuebner** [comment, marked as answer]
Hello @anup-krai,

there currently exists no way to restrict access to environment variables of code running in a Lua sandbox environment.

Kind regards,
Hans

**Reply by anup-krai** [reply]
Hi @hanshuebner,

In the Kong docs [here](https://github.com/Kong/kong/blob/master/kong.conf.default#L1752) it is mentioned that `You can't access or change global values # such as "kong.configuration.pg_password"` but still this variable is accessible in Lua sandboxed environment. 
Please let me know if my understanding is incorrect?

Thanks,
Anup

**Reply by hanshuebner** [reply]
"Global values" in the documentation refers to the the global context of the Lua interpreter, not to process environment variables.
